x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -440,6 +458,23 @@ public synchronized boolean maybeTriggerAsyncJob(long now) {
         return super.maybeTriggerAsyncJob(now);
     }
 
+    /**
+     * Handle new settings at runtime, this is triggered by a call to _transform/id/_update
+     *
+     * @param newSettings The new settings that should be applied
+     */
+    public void applyNewSettings(SettingsConfig newSettings) {
+        auditor.info(transformConfig.getId(), "Transform settings have been updated.");
+        logger.info("[{}] transform settings have been updated.", transformConfig.getId());
+
+        docsPerSecond = newSettings.getDocsPerSecond() != null ? newSettings.getDocsPerSecond() : -1;
+        if (newSettings.getMaxPageSearchSize() != null) {
+            pageSize = newSettings.getMaxPageSearchSize();
```

Comment:
How can the user revert the page size to the default?

Reply:
in case the user sets null, the parsing function sets a "null default", so `newSettings.getMaxPageSearchSize()` will not return null for this case.

However, I think there might still be some bug in here, will investigate.